tests/test_other.py

```diff
@@ -11312,3 +11312,53 @@ def test_hello_function(self):
     # reference them in our docs.  Should we move this file to somewhere else such
     # as `examples/`?)
     self.run_process([EMCC, test_file('hello_function.cpp'), '-o', 'function.html', '-sEXPORTED_FUNCTIONS=_int_sqrt', '-sEXPORTED_RUNTIME_METHODS=ccall,cwrap'])
+
+  def test_es5_transpile(self):
+    # Create a library file that uses the following ES6 features
+    # - let/const
+    # - arrow funcs
+    # - for..of
+    # - object.assign
+    create_file('es6_library.js', '''\
+    mergeInto(LibraryManager.library, {
+      foo: function() {
+        // Object.assign + let
+        let obj = Object.assign({}, {prop:1});
+        err('prop: ' + obj.prop);
+
+        // arror funcs + const
+        const bar = () => 2;
+        err('bar: ' + bar());
+      }
+    });
+    ''')
+    create_file('test.c', 'extern void foo(); int main() { foo(); }')
+    self.emcc_args += ['--js-library', 'es6_library.js']
+    self.uses_es6 = True
+
+    def check_for_es6(expect):
+      js = read_file('test.js')
+      if expect:
+        self.assertContained('() => 2', js)
+        self.assertContained('const ', js)
+        self.assertContained('let ', js)
+      else:
+        self.assertNotContained('() => 2', js)
+        self.assertNotContained('const ', js)
+        self.assertNotContained('let ', js)
+
+    # Check that under normal circumstances none of these features get
+    # removed / transpiled.
+    self.do_runf('test.c', 'prop: 1\nbar: 2\n')
+    check_for_es6(True)
+
+    # If we select and older browser than closure will kick in by default
+    # to traspile.
+    self.emcc_args.remove('-Werror')
+    self.set_setting('MIN_CHROME_VERSION', '10')
+    self.do_runf('test.c', 'prop: 1\nbar: 2\n')
+    check_for_es6(False)
+
+    # If we add `--closure=0` that traspiler (closure) is not run at all
+    self.do_runf('test.c', 'prop: 1\nbar: 2\n', emcc_args=['--closure=0'])
+    check_for_es6(True)
```

Comment:
Please add testing here for the optimization case. That is, that when building with say `-O3` and without and without closure specified (both are fairly common I think) that legacy browser support causes transpilation when we expect and not otherwise.